integrations-and-supported-tools/mosaicml-composer/notebooks/Neptune_MosaicML_Composer.ipynb

```diff
@@ -0,0 +1,393 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "z056VXq1PsFh"
+   },
+   "outputs": [],
+   "source": [
+    "from neptune import ANONYMOUS_API_TOKEN\n",
+    "\n",
+    "neptune_logger = NeptuneLogger(\n",
+    "    api_token=ANONYMOUS_API_TOKEN,  # Replace with your own\n",
+    "    project=\"common/mosaicml-composer\",  # Replace with your own\n",
+    "    capture_hardware_metrics=True,  # This is turned off by default in Notebooks\n",
+    "    tags=[\"mnist\", \"notebook\"],  # (optional) use your own\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "mqvevqK4PsFh"
+   },
+   "source": [
+    "## Configure Composer algorithms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "phLgTbZ0PsFh"
+   },
+   "outputs": [],
+   "source": [
+    "label_smoothing = LabelSmoothing(0.1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "_qtVc6pgPsFi"
+   },
+   "outputs": [],
+   "source": [
+    "prog_resize = ProgressiveResizing(\n",
+    "    initial_scale=0.6,\n",
+    "    finetune_fraction=0.34,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "h4XBrMZIPsFi"
+   },
+   "source": [
+    "## Train model with Composer algorithms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "bgPo-ZAWPsFi"
+   },
+   "outputs": [],
+   "source": [
+    "trainer = Trainer(\n",
+    "    model=mnist_model(),\n",
+    "    train_dataloader=train_dataloader,\n",
+    "    eval_dataloader=eval_dataloader,\n",
+    "    max_duration=\"3ep\",\n",
+    "    device=\"gpu\" if torch.cuda.is_available() else \"cpu\",\n",
+    "    algorithms=[label_smoothing, prog_resize],\n",
+    "    callbacks=ImageVisualizer(),  # Adding the ImageVisualizer() callback automatically logs input images to Neptune\n",
+    "    loggers=neptune_logger,\n",
+    ")\n",
+    "\n",
+    "trainer.fit()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "0p-PoOFSPsFh"
+   },
+   "source": [
+    "**To open the run in the Neptune web app, click the link that appeared in the cell output.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ZIzqE8dDPsFi"
+   },
+   "source": [
+    "## Log additional metadata\n",
+    "`neptune_logger`'s `base_handler` property exposes the [Namespace handler](https://docs.neptune.ai/api/field_types/#namespace-handler) being used by the logger. You can use it to log any additional metadata in the **base namespace used by the logger**.\n",
+    "\n",
+    "Read more about namespaces [in the docs](https://docs.neptune.ai/about/namespaces_and_fields/).\n",
+    "\n",
+    "**Note:** The default base namespace used by `NeptuneLogger` is \"training\". You can update this by passing your own \"base_namespace\" while initializing `NeptuneLogger`.\n",
```

Comment:
Maybe just work this into the cell that creates the logger, or remove it entirely.

Reply:
I wanted to explicitly mention it here as users might otherwise expect to find such metadata logged in the All metadata > sample_image namespace, and not All Metadata > training > sample_image